virtinst/guest.py

```diff
@@ -845,6 +845,37 @@ def _convert_to_vnc_graphics(self):
             dev.add_child(listen)
         dev.set_defaults(self)
 
+    def _convert_to_vnc_video(self):
+        """
+        If there's no video device, add a default one.
+        If there's any qxl device, we rip out every video device and
+            re-add a default one in place
+        We unconditionally disable accel3d if it's enabled.
+        """
+        if not self.devices.video:
+            videodev = DeviceVideo(self.conn)
+            videodev.set_defaults(self)
+            self.add_device(videodev)
+            return
+
+        for video in self.devices.video:
+            # Disable accel3d which is largely tied to spice
+            video.accel3d = None
+
+        qxl_devs = [v for v in self.devices.video if v.model == "qxl"]
+        other_devs = [v for v in self.devices.video if v not in qxl_devs]
+        if not qxl_devs:
+            return
+
+        dev = qxl_devs.pop(0)
+        for video in qxl_devs + other_devs:
+            # We remove other devs, not sure how to deal with mixed setup
+            self.remove_device(video)
```

Comment:
We should not remove all other devices if there are multiple video devices including `qxl`. We can have situation where VM has `bosch` as primary and `qxl` as another video device. This would remove both devices and replaced it with default video device (in most cases `virtio`).

I would change the logic to keep other video devices intact and modify only `qxl` devices. There can be multiple `qxl` devices so if it is `primary` we should replace it with default device, it it's not `primary` we can replace it with `virtio` or remove it completely.

Reply:
Nice, will fix